src/workerd/api/tests/streams-test.js

```diff
@@ -500,6 +471,292 @@ export const finalReadOnInternalStreamReturnsBuffer = {
   },
 };
 
+// Test that canceling a stream rejects body consume function
+export const cancelStreamRejectsBodyConsume = {
+  async test() {
+    const response = new Response('foo bar');
+    const stream = response.body;
+
+    stream.cancel(new Error('a good reason'));
+
+    await rejects(response.text(), TypeError);
+  },
+};
+
+// Test that canceling a reader resolves closed promise
+export const cancelReaderResolvesClosedPromise = {
+  async test() {
+    const response = new Response('foo bar');
+    const stream = response.body;
+    const reader = stream.getReader();
+
+    reader.cancel();
+    const closed = await reader.closed;
+    strictEqual(typeof closed, 'undefined');
+    reader.releaseLock();
+
+    await rejects(response.text(), TypeError);
+  },
+};
+
+// Test that getReader with bad mode throws RangeError
+export const getReaderBadModeThrows = {
+  test() {
+    const response = new Response('foo bar');
+    const stream = response.body;
+
+    throws(() => stream.getReader({ mode: 'nope' }), RangeError);
+  },
+};
+
+// Test that stream is locked after getReader() called
+export const streamLockedAfterGetReader = {
+  test() {
+    const response = new Response('foo bar');
+    const stream = response.body;
+
+    const reader = stream.getReader();
+
+    ok(stream.locked);
+
+    throws(() => stream.getReader(), TypeError);
+
+    reader.releaseLock();
+    ok(!stream.locked);
+    reader.releaseLock(); // Second time should be a no-op
+  },
+};
+
+// Test BYOB reader constraints
+export const byobReaderConstraints = {
+  async test() {
+    const response = new Response('foo bar');
+    const stream = response.body;
+    const reader = stream.getReader({ mode: 'byob' });
+    // Start a read - this will consume part of the stream
+    reader.read(new Uint8Array(32)).catch(() => {}); // Ignore the result
+
+    // Cannot BYOB with a zero-length buffer
+    await rejects(reader.read(new Uint8Array(0)), TypeError);
+
+    // Cannot BYOB an ArrayBuffer, only an ArrayBufferView
+    await rejects(reader.read(new ArrayBuffer(32)), TypeError);
```

Comment:
The internal test case this is replacing checks for a synchronous throw here (which is non-conformant), rather than a rejected promise.

It may be that the internal test case is written using an older compatibility date than what this test is using. If so, I guess we should have tests for both behaviors.

Reply:
This is related to `all-compat-flags` variant of the test we are running. If we want the same code to be executed for both of the cases, we need to have rejects...